### PR TITLE
Health checks

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1113

### CODE REVIEW

File: projects/gloo/pkg/translator/clusters.go [hunk 1]
```diff
@@ -71,6 +75,66 @@ func (t *translator) initializeCluster(upstream *v1.Upstream, endpoints []*v1.En
...
+		}
+		if translatedHc.UnhealthyThreshold == nil {
+			translatedHc.UnhealthyThreshold = defaultThreshold
+		}
+
+		switch healthChecker := hc.GetHealthChecker().(type) {
+		case *v1.HealthCheckConfig_GrpcHealthCheck_:
+			translatedHc.HealthChecker = &envoycore.HealthCheck_GrpcHealthCheck_{
+				GrpcHealthCheck: &envoycore.HealthCheck_GrpcHealthCheck{
+					ServiceName: healthChecker.GrpcHealthCheck.ServiceName,
```
Comment (marcogschmidt) [inline projects/gloo/pkg/translator/clusters.go]: Nit: use getter functions here and below (`GetServiceName()`, etc.).

File: test/e2e/grpc_plugin_test.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/solo-io/gloo/projects/gloo/pkg/defaults"
 )
 
-var _ = Describe("GRPC Plugin", func() {
+var _ = FDescribe("GRPC Plugin", func() {
```
Comment (marcogschmidt) [inline test/e2e/grpc_plugin_test.go]:
```suggestion
var _ = Describe("GRPC Plugin", func() {
```

File: changelog/v0.18.29/fix-e2e-test.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-    description: fix ordering of steps in consul e2e test
-  - type: NON_USER_FACING
-    description: regen solo-kit event loops
-  - dependencyOwner: solo-io
-    dependencyRepo: solo-kit
-    dependencyTag: v0.10.14
-    description: Update solo-kit to version v0.10.14.
-    type: DEPENDENCY_BUMP
\ No newline at end of file
+          - type: FIX
```
Comment (marcogschmidt) [inline changelog/v0.18.29/fix-e2e-test.yaml]: Somehow the indentation got weird here

File: test/e2e/grpc_plugin_test.go [hunk 4]
```diff
@@ -140,6 +140,52 @@ var _ = Describe("GRPC Plugin", func() {
 		}))))
 	})
 
+	It("Grpc Health Checks", func() {
```
Comment (marcogschmidt) [inline test/e2e/grpc_plugin_test.go]: This test is the same as the one before ("Routes to GRPC Functions"), just with the health check config set on the upstream. It seems to me that this isn't actively testing the new feature. Can we maybe update the test to assert that envoy is sending health checks, maybe checking that we get them at the specified interval etc?

File: test/v1helpers/test_grpc_service/testgrpcservice.go [hunk 5]
```diff
@@ -82,3 +92,32 @@ func (s *TestGRPCServer) TestMethod2(_ context.Context, req *glootest.TestReques
 	}()
 	return &glootest.TestResponse2{Str: req.Str}, nil
 }
+
+type healthChecker struct {
```
Comment (EItanya) [inline test/v1helpers/test_grpc_service/testgrpcservice.go]: Will move to go-utils

File: test/services/envoy.go [hunk 6]
```diff
@@ -297,6 +297,11 @@ func (ei *EnvoyInstance) LocalAddr() string {
 	return ei.GlooAddr
 }
 
+func (ei *EnvoyInstance) SetPanicThreshold() error {
+	_, err := http.Post(fmt.Sprintf("http://localhost:%d/runtime_modify?upstream.healthy_panic_threshold=%d", ei.AdminPort, 0), "", nil)
```
Comment (marcogschmidt) [inline test/services/envoy.go]: Does `localhost` here work in both binary and docker modes?

Comment (marcogschmidt) [reply test/services/envoy.go]: Yes, the admin port is exposed with the `-p` docker option.

File: test/e2e/health_checks_test.go [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+
+			Eventually(testRequest, 30, 1).Should(Equal(`{"str":"foo"}`))
+
+			Eventually(tu.C).Should(Receive(PointTo(MatchFields(IgnoreExtras, Fields{
+				"GRPCRequest": PointTo(Equal(glootest.TestRequest{Str: "foo"})),
+			}))))
+		})
+	})
+
+	Context("outlier detection", func() {
```
Comment (marcogschmidt) [inline test/e2e/health_checks_test.go]: Remove?

**Pull request merged**